### ISSUE TITLE
ag cannot find accent letters

### ISSUE THREAD

**Issue by xc1427** [ggreer/the_silver_searcher#1062]
**Version**: [windows port](http://blog.kowalczyk.info/software/the-silver-searcher-for-windows.html) 0.29.1-1641 (built on 2015-07-04)
**System**: windows 7 pro, language français
**Issue**: Ag cannot find french accent letter, such as ô, é, è, ç, à, etc.

**Comment by JFLarvoire** [comment]
Bonjour,  
Having the same problem, I've just done another port of ag for Windows that can find non-ASCII characters... But so far only in text files encoded as UTF-8.  
Also, it will find files that have non-ASCII characters in their pathname.  
The sources are there: https://github.com/JFLarvoire/the_silver_searcher  
If you don't have Visual C++ for rebuilding it, I've uploaded pre-built versions in: http://jf.larvoire.free.fr/temp/ag.zip
Eventually, I plan to add a dynamic detection of the encoding (ANSI/UTF-8/Unicode), to allow finding non-ASCII characters in any kind of text file.  
Jean-François

**Comment by krigstask** [comment]
This should be an issue for [Windows port](https://blog.kowalczyk.info/software/the-silver-searcher-for-windows.html) author.  If it was `ag` issue, it has been fixed already:
```
 % ag ô /tmp/ag.txt  
1:ô, é, è, ç, à
```
(Linux amd64)

**Comment by JFLarvoire** [comment]
Yes, this was indeed a problem in previous Windows ports, not in Unix builds.  
I've released a few days ago a new version 2.0.0 for Windows with full support for this at last:  
https://github.com/JFLarvoire/the_silver_searcher/releases  
As proposed in my previous post above, it dynamically detects the encoding, and can search both in files encoded in UTF-8 and in the Windows System code Page (CP 1252 for west-european versions of Windows).  
And whatever the file encoding, and whatever the current console code page, the strings found will be displayed correctly.
